consensus/reactor.go

```diff
@@ -223,132 +230,148 @@ func (conR *ConsensusReactor) Receive(chID byte, src p2p.Peer, msgBytes []byte)
-			conR.conS.peerMsgQueue <- msgInfo{msg, src.ID()}
-		default:
-			conR.Logger.Error(fmt.Sprintf("Unknown message type %v", reflect.TypeOf(msg)))
-		}
+		case VoteSetBitsChannel:
+			if conR.FastSync() {
+				conR.Logger.Info("Ignoring message received during fastSync", "msg", msg)
+				return
+			}
+			switch msg := msg.(type) {
+			case *VoteSetBitsMessage:
+				cs := conR.conS
+				cs.mtx.Lock()
+				height, votes := cs.Height, cs.Votes
+				cs.mtx.Unlock()
+
+				if height == msg.Height {
+					var ourVotes *cmn.BitArray
+					switch msg.Type {
+					case types.PrevoteType:
+						ourVotes = votes.Prevotes(msg.Round).BitArrayByBlockID(msg.BlockID)
+					case types.PrecommitType:
+						ourVotes = votes.Precommits(msg.Round).BitArrayByBlockID(msg.BlockID)
+					default:
+						panic("Bad VoteSetBitsMessage field Type. Forgot to add a check in ValidateBasic?")
+					}
+					ps.ApplyVoteSetBitsMessage(msg, ourVotes)
+				} else {
+					ps.ApplyVoteSetBitsMessage(msg, nil)
+				}
+			default:
+				// don't punish (leave room for soft upgrades)
+				conR.Logger.Error(fmt.Sprintf("Unknown message type %v", reflect.TypeOf(msg)))
+			}
 
-	case VoteChannel:
-		if conR.FastSync() {
-			conR.Logger.Info("Ignoring message received during fastSync", "msg", msg)
-			return
-		}
-		switch msg := msg.(type) {
-		case *VoteMessage:
-			cs := conR.conS
-			cs.mtx.RLock()
-			height, valSize, lastCommitSize := cs.Height, cs.Validators.Size(), cs.LastCommit.Size()
-			cs.mtx.RUnlock()
-			ps.EnsureVoteBitArrays(height, valSize)
-			ps.EnsureVoteBitArrays(height-1, lastCommitSize)
-			ps.SetHasVote(msg.Vote)
-
-			cs.peerMsgQueue <- msgInfo{msg, src.ID()}
+		case DKGChannel:
+			switch msg := msg.(type) {
+			case *DKGDataMessage:
+				conR.conS.dkg.MsgQueue() <- msgInfo{Msg: msg, PeerID: ""}
+			default:
+				conR.Logger.Error(fmt.Sprintf("Unknown message type %v", reflect.TypeOf(msg)))
+			}
 
 		default:
-			// don't punish (leave room for soft upgrades)
-			conR.Logger.Error(fmt.Sprintf("Unknown message type %v", reflect.TypeOf(msg)))
-		}
-
-	case VoteSetBitsChannel:
-		if conR.FastSync() {
-			conR.Logger.Info("Ignoring message received during fastSync", "msg", msg)
-			return
+			conR.Logger.Error(fmt.Sprintf("Unknown chId %X", chID))
 		}
-		switch msg := msg.(type) {
-		case *VoteSetBitsMessage:
-			cs := conR.conS
-			cs.mtx.Lock()
-			height, votes := cs.Height, cs.Votes
-			cs.mtx.Unlock()
-
-			if height == msg.Height {
-				var ourVotes *cmn.BitArray
-				switch msg.Type {
-				case types.PrevoteType:
-					ourVotes = votes.Prevotes(msg.Round).BitArrayByBlockID(msg.BlockID)
-				case types.PrecommitType:
-					ourVotes = votes.Precommits(msg.Round).BitArrayByBlockID(msg.BlockID)
-				default:
-					panic("Bad VoteSetBitsMessage field Type. Forgot to add a check in ValidateBasic?")
-				}
-				ps.ApplyVoteSetBitsMessage(msg, ourVotes)
-			} else {
-				ps.ApplyVoteSetBitsMessage(msg, nil)
+	} else {
+		switch chID {
+		case DKGChannel:
```

Comment:
It is possible to get rid of code duplication if sending msgInfo{} to MsgQueue will be moved out of switch

Reply:
Fixed.